virttest/virt_vm.py

```diff
@@ -1206,33 +1208,28 @@ def wait_for_serial_login(self, timeout=LOGIN_WAIT_TIMEOUT,
         :params virtio: is a virtio console.
         :return: A ShellSession object.
         """
-        error_messages = []
         logging.debug("Attempting to log into '%s' via serial console "
                       "(timeout %ds)", self.name, timeout)
-        end_time = time.time() + timeout
+        end_time, exception = time.time() + timeout, None
         while time.time() < end_time:
             try:
-                session = self.serial_login(internal_timeout, username,
-                                            password, virtio=virtio)
+                session = self.serial_login(internal_timeout,
+                                            username,
+                                            password,
+                                            virtio=virtio)
                 if restart_network:
-                    os_type = self.params.get("os_type")
-                    try:
-                        logging.debug("Attempting to restart guest network")
-                        utils_net.restart_guest_network(session,
-                                                        os_type=os_type)
-                    except Exception:
-                        pass
+                    logging.debug("Attempting to restart guest network")
+                    os_type = self.params.get('os_type')
+                    utils_net.restart_guest_network(session, os_type=os_type)
                 return session
             except remote.LoginError, e:
+                exception = e
                 self.verify_alive()
-                e = str(e)
-                if e not in error_messages:
-                    logging.debug(e)
-                    error_messages.append(e)
-            time.sleep(2)
```

Comment:
Agree with keep sleep time to 2s. keep error message looks useless I think information we only need to care is exception raised in latest.
Will open a new PR to update wait_for_login function to keep consistence.

Thanks,
Xu